src/Dapr.Actors/Description/InterfaceDescription.cs

```diff
@@ -90,7 +91,7 @@ public MethodDescription[] Methods
             bool useCRCIdGeneration)
         {
             EnsureValidMethods(remotedInterfaceKindName, remotedInterfaceType, methodReturnCheck);
-            var methods = remotedInterfaceType.GetMethods();
+            var methods = remotedInterfaceType.GetInterfaceMethods();
```

Comment:
@eNeRGy164 - can you please separate this commit from this PR and open an issue to discuss. Any behavior changes need to be really well understood. 

I suspect that this is a breaking change we would not accept until the next major release as it *tightens* validation we're performing. We need to talk through all of the possible cases that would be impacted.

Reply:
For example, you said:

> ChildActor has now the description of 3 methods instead of 1 and IOverloadedActor will no longer be seen as valid as GetString is overloaded.

If I understand you correctly, this means that something that is working today will stop working. Even if it's a mistake thats not something we'd just change - these kinds of breaks are very hard for people to reason about when it happens to them.

Reply:
I will create a separate issue for this.

Reply:
Issue #694 created